api/db/migrations/20220517131951_update-constraint-campaign-participations-organizationLearnerId-not-null.js

```diff
@@ -0,0 +1,27 @@
+const CAMPAIGN_PARTICIPATIONS_TABLE = 'campaign-participations';
+const ISIMPROVED_COLUMN = 'isImproved';
+const DELETEDAT_COLUMN = 'deletedAt';
+const DELETEDBY_COLUMN = 'deletedBy';
+const ORGANIZATIONLEARNERID_COLUMN = 'organizationLearnerId';
+const CAMPAIGNID_COLUMN = 'campaignId';
+const NEW_CONSTRAINT_NAME = 'one_active_participation_by_learner';
+
+exports.up = async (knex) => {
+  await knex.schema.alterTable(CAMPAIGN_PARTICIPATIONS_TABLE, function (table) {
+    table.integer(ORGANIZATIONLEARNERID_COLUMN).notNullable().alter();
+  });
+
+  // eslint-disable-next-line knex/avoid-injections
+  return knex.raw(
+    `CREATE UNIQUE INDEX ${NEW_CONSTRAINT_NAME} ON "${CAMPAIGN_PARTICIPATIONS_TABLE}" ("${CAMPAIGNID_COLUMN}", "${ORGANIZATIONLEARNERID_COLUMN}" ) WHERE "${ISIMPROVED_COLUMN}" IS FALSE AND "${DELETEDAT_COLUMN}" IS NULL AND "${DELETEDBY_COLUMN}" IS NULL;`
+  );
+};
```

Comment:
Pour virer   // eslint-disable-next-line knex/avoid-injections
```suggestion
knex.raw(
    `CREATE UNIQUE INDEX :name: ON :participations-table: (:campaignId:, :organizationLearnerId: ) WHERE :isImproved: IS FALSE AND :deletedAt: IS NULL AND :deletedBy: IS NULL;`,
    {
      name: NEW_CONSTRAINT_NAME,
      participations-table: CAMPAIGN_PARTICIPATIONS_TABLE,
      campaignId: CAMPAIGNID_COLUMN,
      organizationLearnerId: ORGANIZATIONLEARNERID_COLUMN,
      isImproved: ISIMPROVED_COLUMN,
      deletedAt: DELETEDAT_COLUMN,
      deletedBy: DELETEDBY_COLUMN,
    }
  );
```

Après les constantes CAMPAIGN_PARTICIPATIONS_TABLE, CAMPAIGNID_COLUMN, ORGANIZATIONLEARNERID_COLUMN, ISIMPROVED_COLUMN, DELETEDAT_COLUMN, DELETEDBY_COLUMN sont utilisées une seule fois 🤷🏽‍♂️